Network/BackendAPIRequest.swift

```diff
@@ -3,7 +3,8 @@ import Foundation
 protocol BackendAPIRequest {
     var endpoint: String { get }
     var method: NetworkService.Method { get }
-    var parameters: [String: AnyObject]? { get }
+    var query: NetworkService.QueryType? { get }
```

Comment:
In what case we don't know whether query type is `JSON` or `Path`? IMO this should be required property to implement.